File: cicodis32/gamelib/full/keen1/controls.cpp
void onKey(int k, int p)
{
    int scancode = 0;
    switch (k)
    {
        case SDL_SCANCODE_SPACE: scancode = 0x39; break;
        case SDL_SCANCODE_UP: scancode = 0x48; break;
        case SDL_SCANCODE_LEFT: scancode = 0x4b; break;
        case SDL_SCANCODE_RIGHT: scancode = 0x4d; break;
        case SDL_SCANCODE_DOWN: scancode = 0x50; break;
        case SDL_SCANCODE_LCTRL: scancode = 0x1d; break;
        case SDL_SCANCODE_1: scancode = 0x1c; break;
        case SDL_SCANCODE_LALT: scancode = 0x38; break;
    }
    if (scancode)
    {
        if (p)
        {
            memoryASet(0x14f2, 0x50ac, scancode);
            memoryASet(0x14f2, 0x50ac, memoryAGet(0x14f2, 0x50ac) | 0x80);
        }
        memoryASet(0x14f2, scancode + 20524, p);
    }
}

void sync()
{
    memoryASet16(ds, 0x5135, memoryAGet16(ds, 0x5135) + 6);
    int t = memoryAGet16(ds, 0x514a);
    if (memoryAGet16(ds, 0x514a) > 0)
    {
        t = t - 1000;
        if (t<0) t = 0;
        memoryASet16(ds, 0x514a, t);
    }
}

File: cicodis32/gamelib/full/icytowerwin32/js/controls.js
document.onkeydown = evt => {
  if (onKey(evt.keyCode, 1))
    evt.preventDefault();
}
document.onkeyup = evt => {
  if (onKey(evt.keyCode, 0))
    evt.preventDefault();
}

function onKey(code, p)
{
  const SDL_SCANCODE_SPACE = 32;
  const SDL_SCANCODE_RETURN = 13;
  const SDL_SCANCODE_LEFT = 37;
  const SDL_SCANCODE_RIGHT = 39;
  const SDL_SCANCODE_UP = 38;
  const SDL_SCANCODE_DOWN = 40;
  const SDL_SCANCODE_LCTRL = 17;
  switch (code)                     
  {
    case SDL_SCANCODE_LEFT: memoryASet(0, memoryAGet32(0, 0x40d1a4)+82, p); break;
    case SDL_SCANCODE_RIGHT: memoryASet(0, memoryAGet32(0, 0x40d1a4)+83, p); break;
    case SDL_SCANCODE_UP: memoryASet(0, memoryAGet32(0, 0x40d1a4)+84, p); break;
    case SDL_SCANCODE_DOWN: memoryASet(0, memoryAGet32(0, 0x40d1a4)+85, p); break;
    case SDL_SCANCODE_RETURN: memoryASet(0, memoryAGet32(0, 0x40d1a4)+67, p); break;
    case SDL_SCANCODE_SPACE: memoryASet(0, memoryAGet32(0, 0x40d1a4)+75, p); break;
    default:
      return false;
  }
  return true;   
}
